### ISSUE TITLE
MD052: Add option to apply to shortcut reference links too 

### ISSUE THREAD

**Issue by waldyrious** [DavidAnson/markdownlint#915]
Currently, [MD052](https://github.com/DavidAnson/markdownlint/blob/v0.29.0/doc/md052.md) has the following caveat:

> "Shortcut" syntax is ambiguous and a missing label will not generate an error. For example, `[shortcut]` could be a shortcut link or the text "shortcut" in brackets.

I'd like to have an option (defaulting to false in light of the caveat above) to make MD052 also apply to shortcut-syntax reference links, so that I could configure it like this:

```yaml
# MD052/reference-links-images
MD052:
  # Also validate reference links using shortcut syntax
  shortcut_links: true
```

_(This issue stems from a discussion that started [here](https://github.com/DavidAnson/markdownlint/issues/456#issuecomment-1645448773).)_

**Comment by DavidAnson** [comment]
This modification would basically assume that every instance of "[whatever]" in a document was intended to be a link rather than text in brackets. I'm not sure that's very common?

**Comment by waldyrious** [comment]
Yeah, I do agree it is likely to catch false positives, hence the suggestion to default to false. My use case is for specific documents, like [this changelog](https://github.com/hamstar/Wikimate/blob/master/CHANGELOG.md?plain=1), where using the [collapsed syntax](https://spec.commonmark.org/0.30/#collapsed-reference-link) would be cumbersome given the number of links. So the configuration to enable this would likely not be project-wise, but file-wise using frontmatter, or even section-wise using HTML comments.

**Comment by nex3** [comment]
I would very much appreciate this. My Markdown corpus universally uses shortcut-style links, because they're strictly shorter than collapsed-style links, but currently we lose all linting for these links (and in practice we relatively frequently forget to add references). I would love a configuration option for MD052 that would catch these issues.

**Comment by waldyrious** [comment]
Thank you for implementing this, @DavidAnson! :100: :pray:

**Comment by ElCuboNegro** [comment]
MD052/reference-links-images is a false positive in all the wikilinks cases.
